service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/NonConsultationResourceMapper.java

```diff
@@ -0,0 +1,180 @@
+import lombok.extern.slf4j.Slf4j;
+import uk.nhs.adaptors.gp2gp.common.service.RandomIdGeneratorService;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.parameters.EncounterTemplateParameters;
+import uk.nhs.adaptors.gp2gp.ehr.utils.CodeableConceptMappingUtils;
+import uk.nhs.adaptors.gp2gp.ehr.utils.TemplateUtils;
+import uk.nhs.adaptors.gp2gp.ehr.utils.XpathExtractor;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.parameters.EncounterTemplateParameters.EncounterTemplateParametersBuilder;
+
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+@Component
+@Slf4j
+public class NonConsultationResourceMapper {
+    private static final Mustache ENCOUNTER_STATEMENT_TO_EHR_COMPOSITION_TEMPLATE =
+        TemplateUtils.loadTemplate("ehr_encounter_to_ehr_composition_template.mustache");
+    private static final String COMPLETE_CODE = "COMPLETE";
+    private static final String DEFAULT_CODE = "<code code=\"196401000000100\" displayName=\"Non-consultation data\" codeSystem=\"2.16"
+        + ".840.1.113883.2.1.3.2.4.15\"/>";
+    private static final String OBSERVATION_COMMENT_CODE = "<code code=\"109341000000100\" displayName=\"GP to GP communication "
+        + "transaction\" codeSystem=\"2.16.840.1.113883.2.1.3.2.4.15\"/>";
+    private static final String BLOOD_PRESSURE_CODE_2 = "<code code=\"109341000000100\" displayName=\"GP to GP communication transaction\""
+        + " codeSystem=\"2.16.840.1.113883.2.1.3.2.4.15\"/>";
+    private static final String DIAGNOSTIC_REPORT_CODE = "<code code=\"109341000000100\" displayName=\"GP to GP communication "
+        + "transaction\" codeSystem=\"2.16.840.1.113883.2.1.3.2.4.15\"/>";
+    private static final String MEDICATION_REQUEST_CODE = "<code code=\"196391000000103\" displayName=\"Non-consultation medication "
+        + "data\" codeSystem=\"2.16.840.1.113883.2.1.3.2.4.15\"/>";
+    private static final String CONDITION_CODE = "<code code=\"109341000000100\" displayName=\"GP to GP communication transaction\" "
+        + "codeSystem=\"2.16.840.1.113883.2.1.3.2.4.15\"/>";
+    private static final String QUESTIONNAIRE_RESPONSE_CODE = "<code code=\"109341000000100\" displayName=\"GP to GP communication "
+        + "transaction\" codeSystem=\"2.16.840.1.113883.2.1.3.2.4.15\"/>";
+    private static final String LOG_TEMPLATE = "Non-consultation resources mapped: {}";
+
+    private final MessageContext messageContext;
+    private final RandomIdGeneratorService randomIdGeneratorService;
+    private final EncounterComponentsMapper encounterComponentsMapper;
+    private final Map<ResourceType, Function<String, EncounterTemplateParameters.EncounterTemplateParametersBuilder>> resourceBuilder =
+        Map.of(
+            ResourceType.Immunization, this::buildImmunization,
+            ResourceType.AllergyIntolerance, this::buildAllergyIntolerance,
+            ResourceType.ReferralRequest, this::buildReferralRequest,
+//          ResourceType.DiagnosticReport, this::buildDiagnosticRequest,
```

Comment:
Why is this commented out? If this is for future work, then adda a `//TODO` with jira reference. The same for commented out line below